main.ts

```diff
@@ -111,6 +147,58 @@ class Translator {
     this.emit(')');
   }
 
+  visitDecorators(decorators: ts.NodeArray<ts.Decorator>) {
+    if (!decorators) return;
+
+    var isAbstract = false, isConst = false;
+    decorators.forEach((d) => {
+      // Special case @CONST & @ABSTRACT
+      // TODO(martinprobst): remove once the code base is migrated to TypeScript.
+      var name = this.ident(d.expression);
+      if (!name && d.expression.kind === ts.SyntaxKind.CallExpression) {
+        var callExpr = (<ts.CallExpression>d.expression);
+        name = this.ident(callExpr.expression);
+      }
+      if (name === 'ABSTRACT') {
+        isAbstract = true;
+        return;
+      }
+      if (name === 'CONST') {
+        isConst = true;
+        return;
+      }
+      if (name === 'IMPLEMENTS') {
+        // Ignore @IMPLEMENTS - it's handled above in visitClassLike.
+        return;
+      }
+      this.emit('@');
+      this.visit(d.expression);
+    });
+    if (isAbstract) this.emit('abstract');
```

Comment:
could emit inside the loop, unless we are really concerned that CONST or ABSTRACT appears more than once